openwisp-config/files/openwisp.agent

```diff
@@ -133,11 +140,23 @@ register() {
 	local model=$(cat /tmp/sysinfo/model)
 	local os=$(ubus call system board | grep description | awk -F\" '{ print $4 }')
 	local system=$(ubus call system board | grep system | awk -F\" '{ print $4 }')
+	# get hardware id if script is given
+	if [ -x "$HARDWARE_ID_SCRIPT" ]; then
+		local hardware_id=$("$HARDWARE_ID_SCRIPT")
+	else
+		local hardware_id=""
```

Comment:
this one kind of puzzles me, why do we need to have this variable set to an empty string in this case?

Reply:
At first I added the `hardware_id` to the `FETCH_COMMAND` in https://github.com/TDT-AG/openwisp-config/blob/tdt-hardware-id/openwisp-config/files/openwisp.agent#L166 only if `HARDWARE_ID_SCRIPT` is set and executable.
But I was not convinced of my code for that and figured that it wouldn't hurt to add `hardware_id` in any case because an empty value gets converted to `None` in `django-netjsonconfig`.